### DISCUSSION TITLE
Moving to Paras (In multiple batches?)

### DISCUSSION THREAD

**Discussion by SeHarlan** [TENK-DAO/tenk#28] [Q&A] [answered]
Hey all, I'm beginning to work on a project that I would like to mint in batches, yet still have reveal functionality per batch. 
Ideally I would be able to move each batch over to paras after reveal. Then when the time comes do another reveal (on the same contract) and port that to Paras.

Does anyone know if this is possible, or would I have to wait till all batches are revealed and then move them over to paras?

I've also asked this in the Paras TG so If I hear back there first I'll post the response here in case someone else ends up having the same question.

**Comment by willemneal** [comment, marked as answer]
Currently TenK assumes a range of token ids. E.g. 0-9,999.  This is done through a raffle data structure with a uri that points to a directory with assets numbered in that range. That can be the initial release.  Then after those are minted, you can update the uri to include the initial assets and then 10,000-19,999, and reset the raffle with this new range.  So yes this is possible with the same contract.

As far as paras is concerned you just need the contract whitelisted.  So after updating for the subsequent batches they will appear as they are minted.

**Reply by SeHarlan** [reply]
Awesome, thank you!
